src/components/dashboard/menu-table/menu-table.js

```diff
@@ -284,16 +287,27 @@ const MenuTable = (props) => {
         <>
             <MenuControls>
                 <form onSubmit={handleSearch} className='searchForm'>
-                    <input className='search' placeholder='Search Dishes...' id='searchBox' type='text' value={searchBoxValue} onChange={(e) => setSearchBoxValue(e.target.value)} />
+                    <input className='search' placeholder='Search Dishes...' id='searchBox' type='text' value={searchBoxValue} 
+                        onChange={(e) => setSearchBoxValue(e.target.value)} 
+                        onFocus={(e) => {
+                            setSearchBoxFocused(true); 
+                            e.target.select(); // highlight text when focus on element
+                            setAfterSubmit(false);
+                        }} 
+                        onBlur={(e) => setSearchBoxFocused(false)}
+                    />
                     {
-                        (isSearching && searchBoxValue != null) ?
+                        (afterSubmit && searchBoxValue != '' && !searchBoxFocused) ?
                         <input className='cancelSearch' type='image' alt="Reset search" src={CancelIcon} onClick={(e) => {
                             e.preventDefault();
                             setSearchBoxValue('');
                             setIsSearching(false);
-                        }}/> : null
+                            setAfterSubmit(false);
+                        }}/> : 
+                        <input className='submitSearch' type='image' alt="Submit" src={SearchIcon} onClick={(e) => {
+                            setAfterSubmit(true);
+                        }}/>
```

Comment:
Hmmm I remember you could exit searching mode when you hit enter with an empty search box. How does someone exit search now when they clear the search box by hitting backspace? I'm also having trouble following afterSubmit and isSearching...

Reply:
Sorry so many variables, let me try to explain.

The customer can exit search by either pressing enter or magnifying glass icon with empty text box OR by pressing X icon while there's text in the box.

isSearching - changes to true after the search is submitted the first time, user can continue searching and isSearching doesn't turn back to false until after they have exited the search (X icon or submitting with empty text box). This variable is used to tell if search results table should be displayed or normal menu should be displated.

afterSubmit - changes to true after the search is submitted, changes back to false when the user enters the text box again to make another search. This variable is used to tell if we should display magnifying glass icon or X icon (afterSubmit = true then show X icon, afterSubmit = false show magnifying glass icon)

These two only differ when a user wants to execute multiple searches in a row without exiting the search in between. Let me know if that makes sense and if I answered your question.

Reply:
Looks like when the user presses submit with empty box doesn't work anymore to exit the search, I will fix this

Reply:
Also, writing this explanation made me realize I could just use a combination of isSearching and searchBoxFocused and get rid of afterSubmit (: